napari/_qt/widgets/qt_highlight_preview.py

```diff
@@ -496,8 +498,8 @@ def setValue(self, value: dict):
         value : dict
             Highlight value (thickness and color).
         """
-        self._update_thickness_value(value['highlight_thickness'])
-        self._update_color_value(value['highlight_color'])
+        self._update_thickness_value(value.get('highlight_thickness', ''))
+        self._update_color_value(value.get('highlight_color', ''))
```

Comment:
When is this raise exception?

Reply:
A possible exception can be raised now due to the changes I did to support updating preference widgets from nested settings at https://github.com/napari/napari/blob/338a07faec7cfa81be04ebc3fa491f2b8e4a57f5/napari/_qt/dialogs/preferences_dialog.py#L142-L155 so now there is a possibility to get settings updates from highlight attributes individually (so updating only the highlight color value or the highlight thicknes). So basically, although the value that is passed to do the update is always a dictionary, the dictionary could only have one of the two attibutes